### PR TITLE
Fetch JITM on pull to refresh in MyStore screen

### LINKED ISSUE #7926: Fetch JITMs when pull to refreshed in the MyStore screen
JITMs are fetched only on the fresh app start. This works most of the time. However, there could be certain scenarios where the user might miss the new JITMs or the currently displayed JITM might be stale if they have kept the app in the memory for a longer time.

Fetching JITM on the pull to refresh should solve the above-mentioned issues to some extent.

### PR DESCRIPTION
**Pull request by AnirudhBhat**
Closes: #7926 

### Description

JITMs are fetched only on the fresh app start. This works most of the time. However, there could be certain scenarios where the user might miss the new JITMs or the currently displayed JITM might be stale if they have kept the app in the memory for a longer time.

Fetching JITM on the pull to refresh should solve the above-mentioned issues to some extent. This PR fetches JITMs when pull to refresh in the MyStore screen.

#### Pre-requisite
As mentioned in this post (pdfdoF-1vt-p2) make sure you have installed any one of the mentioned barcode plugin installed on your store. For example - you can install [A4 Barcode Generator](https://wordpress.org/plugins/a4-barcode-generator/)

Your store must be set up in the US. JITM is not supported in any other country.

1. Visit `MyStore` screen
2. Ensure you see the banner
3. Pull to refresh in the `MyStore` screen
4. Check logs to see the track events being fired again for fetching (JITM_FETCH_SUCCESS) and displaying the JITM (JITM_DISPLAYED)
5. You can also check in flipper to ensure the network call for fetching JITM is being made after pull to request.

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/src/test/kotlin/com/woocommerce/android/ui/mystore/MyStoreViewModelTest.kt [hunk 1]
```diff
@@ -418,6 +419,20 @@ class MyStoreViewModelTest : BaseUnitTest() {
         }
 
     // region Just In Time Messages (JITM)
+    @Test
+    fun `Given network connection, When on swipe to refresh, then fetch JITMS`() {
```
Comment (kidinov) [inline WooCommerce/src/test/kotlin/com/woocommerce/android/ui/mystore/MyStoreViewModelTest.kt]: np: `G` is capital

**Review by kidinov — COMMENTED** [review]
LGTM!

Left a np. Also, have you considered calling it from "onResume"? Any downsides of that approach?

**Comment by AnirudhBhat** [comment]
>have you considered calling it from "onResume"? Any downsides of that approach?

Good question! I did consider this approach before going with the pull-to-refresh approach. The downside is that `onResume` might happen far too frequently than manual pull-to-refresh. JITM won't change very frequently and the API call we make on every `onResume` will be inefficient and will be hitting the server for not much gain. 

The scenario we are trying to avoid with pull-to-refresh is when the app lives in the memory for a long time and when the user comes back, the JITM might be irrelevant for them or they might miss a new JITM until they kill and open the app again. This shouldn't happen very often. But when it happens, pull-to-refresh should fix this.

File: WooCommerce/src/test/kotlin/com/woocommerce/android/ui/mystore/MyStoreViewModelTest.kt (hunk 1 above)
Comment (AnirudhBhat) [reply WooCommerce/src/test/kotlin/com/woocommerce/android/ui/mystore/MyStoreViewModelTest.kt]: Fixed the naming convention in the entire file here: https://github.com/woocommerce/woocommerce-android/pull/7938/commits/db5e01b74db0967edc91702a07edc446dc610bb6

**Review by kidinov — APPROVED** [review]

**Pull request merged**